### PR TITLE
v.tin.to.rast: Python fixes

### PR DESCRIPTION
**Pull request by kikislater**
https://github.com/OSGeo/grass-addons/issues/515

### CODE REVIEW

**Comment by nilason** [comment]
You just beat me to it :) ! I was just about to suggest following changes, which 1) removes `nuldev` as a (unused and unnecessary) global variable; 2) make sure the opened "files" are properly closed (I would consider this good practice to close what is opened, even though they would be closed anyway at the end of the script); and (3) make the changed lines Black formatted.
 
```diff
--- grass7/vector/v.tin.to.rast/v.tin.to.rast.py.orig	2021-04-27 20:41:55.000000000 +0200
+++ grass7/vector/v.tin.to.rast/v.tin.to.rast.py	2021-04-27 20:44:08.000000000 +0200
@@ -60,17 +60,18 @@
     from grass.lib.vector import *
     from grass.lib.raster import *
 
+
 def cleanup():
-    nuldev = file(os.devnull, 'w')
     if tmp:
-        grass.run_command('g.remove', type = 'raster',
-                          name = '%s' % tmp,
-                          quiet = True, stderr = nuldev)
+        with open(os.devnull, "w") as nuldev:
+            grass.run_command(
+                "g.remove", type="raster", name="%s" % tmp, quiet=True, stderr=nuldev
+            )
+
 
 def main():
 
-    global nuldev, tmp
-    nuldev = file(os.devnull, 'w')
+    global tmp
     tmp = "v_tin_to_rast_%d" % os.getpid()
 
 
@@ -165,12 +166,19 @@
                                  map = input).rsplit()[5].split('=')[1]
 
     tmp = "v_tin_to_rast_%d" % os.getpid()
-    grass.mapcalc("$tmp = if($vbottom < $output && $output < $vtop, $output, null())",
-                  tmp = tmp, output = output, vbottom = vbottom, vtop = vtop,
-                  quiet = True, stderr = nuldev)
+    with open(os.devnull, "w") as nuldev:
+        grass.mapcalc(
+            "$tmp = if($vbottom < $output && $output < $vtop, $output, null())",
+            tmp=tmp,
+            output=output,
+            vbottom=vbottom,
+            vtop=vtop,
+            quiet=True,
+            stderr=nuldev,
+        )
 
-    grass.parse_command('g.rename', rast = (tmp, output),
-                      quiet = True, stderr = nuldev)
+    with open(os.devnull, "w") as nuldev:
+        grass.parse_command("g.rename", rast=(tmp, output), quiet=True, stderr=nuldev)
 
     # write cmd history:
     grass.run_command('r.support', map = output,
```

**Comment by nilason** [comment]
Alternatively (and probably better) keeping `nuldev` as ("real") global variable and closing it explicitly in `cleanup()` with: 

```diff
--- grass7/vector/v.tin.to.rast/v.tin.to.rast.py.orig	2021-04-27 20:41:55.000000000 +0200
+++ grass7/vector/v.tin.to.rast/v.tin.to.rast.py	2021-04-27 20:44:08.000000000 +0200
@@ -61,11 +61,13 @@
     from grass.lib.raster import *
 
 def cleanup():
-    nuldev = file(os.devnull, 'w')
     if tmp:
-        grass.run_command('g.remove', type = 'raster',
-                          name = '%s' % tmp,
-                          quiet = True, stderr = nuldev)
+        grass.run_command(
+            "g.remove", type="raster", name="%s" % tmp, quiet=True, stderr=nuldev
+        )
+
+    nuldev.close()
+
 
 def main():
 
@@ -165,12 +167,17 @@
                                  map = input).rsplit()[5].split('=')[1]
 
     tmp = "v_tin_to_rast_%d" % os.getpid()
-    grass.mapcalc("$tmp = if($vbottom < $output && $output < $vtop, $output, null())",
-                  tmp = tmp, output = output, vbottom = vbottom, vtop = vtop,
-                  quiet = True, stderr = nuldev)
+    grass.mapcalc(
+        "$tmp = if($vbottom < $output && $output < $vtop, $output, null())",
+        tmp=tmp,
+        output=output,
+        vbottom=vbottom,
+        vtop=vtop,
+        quiet=True,
+        stderr=nuldev,
+    )
 
-    grass.parse_command('g.rename', rast = (tmp, output),
-                      quiet = True, stderr = nuldev)
+    grass.parse_command("g.rename", rast=(tmp, output), quiet=True, stderr=nuldev)
 
     # write cmd history:
     grass.run_command('r.support', map = output,
```

File: grass7/vector/v.tin.to.rast/v.tin.to.rast.py [hunk 1]
```diff
@@ -61,16 +61,17 @@
...
+            "g.remove", type="raster", name="%s" % tmp, quiet=True, stderr=nuldev
+        )
+
+    nuldev.close()
 
 def main():
 
     global nuldev, tmp
-    nuldev = file(os.devnull, 'w')
+    nuldev = open(os.devnull, 'w')
```
Comment (nilason) [inline grass7/vector/v.tin.to.rast/v.tin.to.rast.py:74]:
```suggestion
    nuldev = open(os.devnull, "w")
```

Comment (nilason) [reply grass7/vector/v.tin.to.rast/v.tin.to.rast.py:74]: Minor detail to appease Black formatting rules.

**Review by nilason — APPROVED** [review]

File: grass7/vector/v.tin.to.rast/v.tin.to.rast.py (hunk 1 above)
Comment (kikislater) [reply grass7/vector/v.tin.to.rast/v.tin.to.rast.py]: That's why you mean by black formatting, sorry I didn't get you ;-)

**Comment by nilason** [comment]
Sorry for taking things for obvious (when it's not!). By Black formatting see [Black](https://black.readthedocs.io/en/stable/) and for some GRASS related background on this see [mailing list thread](https://lists.osgeo.org/pipermail/grass-dev/2021-February/094993.html).

**Comment by kikislater** [comment]
> Sorry for taking things for obvious (when it's not!). By Black formatting see [Black](https://black.readthedocs.io/en/stable/) and for some GRASS related background on this see [mailing list thread](https://lists.osgeo.org/pipermail/grass-dev/2021-February/094993.html).

No worries thank you for details, first time I heard about black !

**Comment by nilason** [comment]
Thanks!

**Comment by kikislater** [comment]
> Thanks!

;-) I didn't do a lot, I learnt more from you than I fixed. But now extension works again ! Thanks

**Pull request merged**